aiida/cmdline/commands/cmd_group.py

```diff
@@ -84,6 +84,70 @@ def group_remove_nodes(group, nodes, clear, force):
         group.remove_nodes(nodes)
 
 
+@verdi_group.command('move-nodes')
+@options.SOURCE_GROUP(required=True)
+@options.TARGET_GROUP(required=True)
+@arguments.NODES()
+@options.FORCE(help='Do not ask for confirmation and skip all checks.')
+@with_dbenv()
+def group_move_nodes(source_group, target_group, nodes, force):
+    """Move nodes from one group to another."""
+    from aiida.orm import Group, Node, QueryBuilder
+
+    source_label = source_group.label
+    source_class = source_group.__class__.__name__
+
+    target_label = target_group.label
+    target_class = target_group.__class__.__name__
+
+    if not force:
+
+        if source_label == target_label:
+            echo.echo_critical(f'Source and target group are the same: {source_class}<{source_label}>.')
+
+        if nodes:
+            node_pks = [node.pk for node in nodes]
+
+            query = QueryBuilder()
+            query.append(Group, filters={'label': source_label}, tag='group')
+            query.append(Node, with_group='group', filters={'id': {'in': node_pks}}, project='id')
+
+            source_group_node_pks = query.all(flat=True)
+
+            if not source_group_node_pks:
+                echo.echo_critical(f'None of the specified nodes are in {source_class}<{source_label}>.')
+
+            if len(node_pks) > len(source_group_node_pks):
+                absent_node_pks = set(node_pks).difference(set(source_group_node_pks))
+                echo.echo_warning(
+                    f'{len(absent_node_pks)} nodes with PK {absent_node_pks} are not in {source_class}<{source_label}>.'
+                )
+                nodes = [node for node in nodes if node.pk in source_group_node_pks]
+
+            query = QueryBuilder()
+            query.append(Group, filters={'label': target_label}, tag='group')
+            query.append(Node, with_group='group', filters={'id': {'in': node_pks}}, project='id')
+
+            target_group_node_pks = query.all(flat=True)
+
+            if target_group_node_pks:
+                echo.echo_warning(
+                    f'{len(target_group_node_pks)} nodes with PK {set(target_group_node_pks)} are already in '
+                    f'{target_class}<{target_label}>. These will still be removed from {source_class}<{source_label}>.'
+                )
+
+            click.confirm(
+                f'Are you sure you want to move {len(nodes)} nodes from {source_class}<{source_label}> '
+                f'to {target_class}<{target_label}>?',
+                abort=True
+            )
+        else:
+            echo.echo_critical('No nodes identifiers were provided.')
```

Comment:
Should this be required? Shouldn't it be possible to move _all_ nodes from source to another group?

Reply:
If you want to keep it, I would invert the logical clause and just have `if not nodes: echo_critical`. Then you can dedent all the other logic making it easier to read.

Reply:
Fair point, moving all nodes to another group is a valid use case that currently isn't covered. I'm wondering if:

1. This should be the behaviour if no nodes are specified.
2. The user has to add e.g. `-a` / `--all` to move all nodes.

Perhaps [2] is better, because if the user accidentally already added `-f` and then presses enter without specifying a node, it'll be annoying to move the nodes back again.

In case neither nodes nor `--all` are specified, the command should perhaps also just do nothing. That's also what happens when you do e.g. `verdi process show` without specifying any nodes.

Reply:
Yeah you are probably right that making it explicit with an option is the safest option